src/corelibs/U2Core/src/models/GObject.cpp

```diff
@@ -172,18 +172,10 @@ void GObject::setRelationsInDb(QList<GObjectRelation> &list) const {
         const U2DataType refType = U2ObjectTypeUtils::toDataType(relation.ref.objType);
         const bool relatedObjectDbReferenceValid = relation.ref.entityRef.dbiRef.isValid();
 
-        if (U2Type::Unknown == refType || (relatedObjectDbReferenceValid && !(relation.ref.entityRef.dbiRef == entityRef.dbiRef))) {
+        if (U2Type::Unknown == refType || !relatedObjectDbReferenceValid) {
             continue;
         }
 
-        if (!relatedObjectDbReferenceValid) {
```

Comment:
This method had a contract before: is relation object has invalid dbi-ref we will search for the valid target by name.
Why you decided to remove this behavior? I may affect other logic not related to your bug